### PR TITLE
Delete some redundant code

### PR DESCRIPTION
**Pull request by xichengliudui**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Delete some redundant code

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/client-go/tools/cache/heap.go [hunk 1]
```diff
@@ -251,9 +251,8 @@ func (h *Heap) Pop() (interface{}, error) {
 	obj := heap.Pop(h.data)
 	if obj != nil {
 		return obj, nil
-	} else {
-		return nil, fmt.Errorf("object was removed from heap data")
 	}
+	return nil, fmt.Errorf("object was removed from heap data")
```
Comment (miguelbernadi) [inline staging/src/k8s.io/client-go/tools/cache/heap.go]: It's probably safe to compare `obj` against nil as it should be a pointer, not an interface. In that case we can swap the error flow into the check and leave the valid return in the main flow, to align the happy path with the least indentations.

**Comment by xichengliudui** [comment]
@miguelbernadi ，I have changed.

**Pull request merged**